src/main/java/com/octopus/bamboo/plugins/task/octopusmetadata/OctopusMetadataTask.java

```diff
@@ -54,18 +56,35 @@ public OctopusMetadataTask(@NotNull @ComponentImport final ProcessService proces
     public TaskResult execute(@NotNull final CommonTaskContext taskContext) throws TaskException {
         checkNotNull(taskContext, "taskContext cannot be null");
 
-        final String serverUrl = taskContext.getConfigurationMap().get(OctoConstants.SERVER_URL);
-        final String apiKey = taskContext.getConfigurationMap().get(OctoConstants.API_KEY);
-        final String spaceName = taskContext.getConfigurationMap().get(OctoConstants.SPACE_NAME);
-        final String forceUpload = taskContext.getConfigurationMap().get(OctoConstants.FORCE);
-        final Boolean forceUploadBoolean = BooleanUtils.isTrue(BooleanUtils.toBooleanObject(forceUpload));
-        final String loggingLevel = taskContext.getConfigurationMap().get(OctoConstants.VERBOSE_LOGGING);
+        ConfigurationMap configurationMap = taskContext.getConfigurationMap();
+
+        final String serverUrl = configurationMap.get(OctoConstants.SERVER_URL);
+        final String apiKey = configurationMap.get(OctoConstants.API_KEY);
+        final String spaceName = configurationMap.get(OctoConstants.SPACE_NAME);
+
+        final String overwriteModeString = configurationMap.get(OctoConstants.OVERWRITE_MODE);
+        OverwriteMode overwriteMode = OverwriteMode.FailIfExists;
+
+        // if we don't have a overwriteMode defined then the step still has "legacy data" for forceUpload.
+        if (StringUtils.isEmpty(overwriteModeString)) {
+            final String forceUpload = configurationMap.get(OctoConstants.FORCE);
+            final Boolean forceUploadBoolean = BooleanUtils.isTrue(BooleanUtils.toBooleanObject(forceUpload));
+
+            if (forceUploadBoolean) {
+                overwriteMode = OverwriteMode.OverwriteExisting;

```

Comment:
This would be slightly clearer as a ternary, rather than a conditional change to a value initialized at a distance. Ideally then the initial value could disappear, but I can't remember if Java supports that like C#.

Reply:
this works
```
overwriteMode = forceUploadBoolean ? OverwriteMode.OverwriteExisting : OverwriteMode.valueOf(overwriteModeString); 
```